### PR TITLE
Allow CALL_CLOSED to be raised when call_id is not set.

### PR DESCRIPTION
**Pull request by trampster**
Fix for #747

### CODE REVIEW

File: modules/ctrl_tcp/ctrl_tcp.c [hunk 1]
```diff
@@ -275,7 +275,10 @@ static void ua_event_handler(struct ua *ua, enum ua_event ev,
 	err = odict_entry_add(od, "event", ODICT_BOOL, true);
 	err |= event_encode_dict(od, ua, ev, call, prm);
 	if (err)
+	{
```
Comment (alfredh) [inline modules/ctrl_tcp/ctrl_tcp.c]: please move curly bracket to previous line, like this:

```c
if (err) {
}
```

File: src/event.c [hunk 2]
```diff
@@ -129,13 +129,21 @@ int event_encode_dict(struct odict *od, struct ua *ua, enum ua_event ev,
...
 		dir = call_is_outgoing(call) ? "outgoing" : "incoming";
 
 		err |= odict_entry_add(od, "direction", ODICT_STRING, dir);
 		err |= odict_entry_add(od, "peeruri",
 				       ODICT_STRING, call_peeruri(call));
-		err |= odict_entry_add(od, "id", ODICT_STRING, call_id(call));
+
+		call_identifier = call_id(call);
+		if (call_identifier)
+		{
```
Comment (alfredh) [inline src/event.c]: please move curly bracket to previous line
(same as previous comment).

**Comment by alfredh** [comment]
thanks for the PR, this looks very good.

can you also take a look @jmillan ?

File: modules/ctrl_tcp/ctrl_tcp.c (hunk 1 above)
Comment (trampster) [reply modules/ctrl_tcp/ctrl_tcp.c]: done

File: src/event.c (hunk 2 above)
Comment (trampster) [reply src/event.c]: done

**Comment by jmillan** [comment]
LGTM 👍

**Comment by trampster** [comment]
Why does it say 'Some checks haven't complete yet', its been 11 hours and clicking on Details shows ticks against everything.

**Comment by alfredh** [comment]
not sure about travis .. anyhow merged now .. thank you !

**Pull request merged**